### PR TITLE
Implement support for CAN-FD and container frames

### PR DESCRIPTION
**Pull request by andlaus**
this is a direct carryover of the [same PR](https://github.com/eclipse/kuksa.val.feeders/pull/151) from the old repository. All discussions for the original PR still apply.

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: dbcfeederlib/canreader.py [hunk 1]
```diff
@@ -83,44 +89,58 @@ def stop(self):
...
-                            )
-        except Exception:
-            log.warning("Error processing CAN message with frame ID: %#x", frame_id, exc_info=True)
+        message_def = self._mapper.get_message_for_canid(frame_id)
+        if message_def is not None:
+            try:
+                decode = message_def.decode(bytes(data), allow_truncated=True, decode_containers=True)
+            except Exception as e:
+                log.warning("Error processing CAN message with frame ID: %#x", frame_id, exc_info=True)
+                log.warning("Error:  ", e)
```
Comment (erikbosch) [inline dbcfeederlib/canreader.py:98]: This line gives a linter error (i think), duplicate as we already have the line above?

**Comment by erikbosch** [comment]
Thanks for moving the PR. The eclipsefdn/eca check fails, in one of your commits `alwalz1@cmtclmlvxqr6x4k.rd.corpintra.net` is listed as author and that user has no ECA. I do not know if the ECA check is run only once (until it succeeds) and that is the reason why we did not got an error for that commit earlier in the old repo. 

Maybe it is possibly for you to edit that commit so that it gets the same author as other commits


```
commit c3d38eb0b3a4ffe7c746218a5b3e22c4a7ce3ea1 (HEAD -> pr6)
Author: Walz <alwalz1@cmtclmlvxqr6x4k.rd.corpintra.net>
Date:   Tue Nov 7 15:30:34 2023 +0100

    Make flake8 happy :-)
    
    Signed-off-by: Alexander Walz <alexander.walz@mbition.io>
    Signed-off-by: Andreas Lauser <andreas.lauser@mbition.io>

commit d3b05c6f55e54a2a4da11434be2e5413a0097c01
Author: Alexander Walz <alexander.walz@mercedes-benz.com>
Date:   Fri Nov 3 09:53:56 2023 +0100

    Adapt output message for reading the database file
    
    Signed-off-by: Alexander Walz <alexander.walz@mbition.io>
    Signed-off-by: Andreas Lauser <andreas.lauser@mbition.io>
```

**Comment by andlaus** [comment]
>  The eclipsefdn/eca check fails, in one of your commits alwalz1@cmtclmlvxqr6x4k.rd.corpintra.net is listed as author and that user has no ECA. 

oops. git configuration error. fixed...

File: dbcfeederlib/canreader.py (hunk 1 above)
Comment (andlaus) [reply dbcfeederlib/canreader.py:98]: how can this be reproduced locally? on my machine, `flake8` passes fine:
```
$ flake8 .
$
```

Comment (erikbosch) [reply dbcfeederlib/canreader.py:98]: This is not reported by flake8, it is `pylint` that we run from https://github.com/eclipse-kuksa/kuksa-can-provider/blob/main/.github/workflows/can-provider.yml#L122

Installed in CI by

`pip3 install --no-cache-dir -r requirements.txt -r requirements-dev.txt`

This is how I run it locally:

```
erik@debian4:~/kuksa-can-provider$ pylint -E dbcfeeder.py dbcfeederlib test
************* Module dbcfeederlib.canreader
dbcfeederlib/canreader.py:98:16: E1205: Too many arguments for logging format string (logging-too-many-args)
```

**Comment by erikbosch** [comment]
The remaining pre-commit error might be a false positive. I think I have seen a similar problem in another repo and will update dependencies and retry

**Comment by andlaus** [comment]
> The remaining pre-commit error might be a false positive. I think I have seen a similar problem in another repo and will update dependencies and retry

yes, seems like flake8 does plays style police even within strings. (this is a bit weird IMO, even more so considering that the flake8 on my computer does not complain about it.) anyway, I've added a space before the `x` format...

**Review by erikbosch — APPROVED** [review]

**Pull request merged**